builder/scaleway/step_pre_validate.go

```diff
@@ -0,0 +1,76 @@
+package scaleway
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/packer/helper/multistep"
+	"github.com/hashicorp/packer/packer"
+	"github.com/scaleway/scaleway-sdk-go/api/instance/v1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+// StepPreValidate provides an opportunity to pre-validate any configuration for
+// the build before actually doing any time consuming work
+//
+type stepPreValidate struct {
+	Force        bool
+	ImageName    string
+	SnapshotName string
+}
+
+func (s *stepPreValidate) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
+	ui := state.Get("ui").(packer.Ui)
+
+	if s.Force {
+		ui.Say("Force flag found, skipping prevalidating image name")
+		return multistep.ActionContinue
+	}
+
+	ui.Say(fmt.Sprintf("Prevalidating image name: %s", s.ImageName))
+
+	instanceAPI := instance.NewAPI(state.Get("client").(*scw.Client))
+	images, err := instanceAPI.ListImages(&instance.ListImagesRequest{})
```

Comment:
You could add a Name filter here, and add `scw.WithAllPages()` as a second argument

Reply:
the `Name` field of `ListImagesRequest` is undocumented, both in the Go [source code](https://github.com/scaleway/scaleway-sdk-go/blob/0df3fbac75bc869bd0c841fe0c85a401bc7450e3/api/instance/v1/instance_sdk.go#L1965) and in the [HTTP API documentation](https://developers.scaleway.com/en/products/instance/api/#get-5f69ed).

Looking at the documentation of other `Name` filters in the code, sometimes it gives the impression that the filter is equivalent to `strings.Contains()`.

What should I set `Name` to ? To the exact image name ? That would also allow to avoid the loop.

Reply:
Yep it's a fuzzi match so it's kinda like a `strings.Contains()`. It doesn't remvove the loop but it allows to get fewer possibel results :smile: and yep Name to exact image name

Reply:
Ok thanks.

Reply:
done